src/main/java/org/cbioportal/security/VirtualStudyPermissionService.java

```diff
@@ -0,0 +1,56 @@
+package org.cbioportal.security;
+
+import org.cbioportal.utils.security.AccessLevel;
+import org.cbioportal.web.parameter.StudyViewFilter;
+import org.cbioportal.web.parameter.VirtualStudy;
+import org.cbioportal.web.parameter.VirtualStudyData;
+import org.cbioportal.web.parameter.VirtualStudySamples;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class VirtualStudyPermissionService {
+
+    private final Optional<CancerStudyPermissionEvaluator> cancerStudyPermissionEvaluator;
+
+    public VirtualStudyPermissionService(Optional<CancerStudyPermissionEvaluator> cancerStudyPermissionEvaluator) {
+        this.cancerStudyPermissionEvaluator = cancerStudyPermissionEvaluator;
+    }
+
+    public void filterOutForbiddenStudies(List<VirtualStudy> virtualStudies) {
```

Comment:
@haynescd @pieterlukasse let's decide whether we should filter out forbidden studies/samples at all.